### ISSUE TITLE
DOC: update the pyproj transform example for deprecated syntax

### ISSUE THREAD

**Issue by jorisvandenbossche** [shapely/shapely#826]
The example using pyproj in `transform` (https://github.com/Toblerity/Shapely/pull/783) is now raising deprecation warnings with the latest pyproj release:

```
In [2]: pyproj.Proj(init='epsg:4326')
/home/joris/miniconda3/envs/geo-dev/lib/python3.8/site-packages/pyproj/crs.py:77: FutureWarning: '+init=<authority>:<code>' syntax is deprecated. '<authority>:<code>' is the preferred initialization method.
  return _prepare_from_string(" ".join(pjargs))
Out[2]: Proj('+proj=longlat +datum=WGS84 +no_defs', preserve_units=True)
```

So we should update that to use it without `init=` (and could maybe also use `CRS` instead)

**Comment by tomplex** [comment]
`pyproj` 2.1 actually introduced new functionality to do this with the [`Transformer`](https://pyproj4.github.io/pyproj/stable/api/transformer.html) API, and specifically calls out using `Proj` in the [gotcha's section](https://pyproj4.github.io/pyproj/stable/gotchas.html#proj-not-a-generic-latitude-longitude-to-projection-converter) as not the right thing when you're converting geographic coords to projected coords (which, for me at least, is almost always the case). `Transformer` seems to be the preferred API going forward. Since many folks are probably copying the example right from the docs, should they be updated to use this newer `Transformer` API instead?

**Comment by sgillies** [comment]
I'm out of time to work on this. Would be grateful for a PR from a current pyproj user 🙏

**Comment by tomplex** [comment]
Released in 1.7.1

**Comment by jorisvandenbossche** [comment]
This was fixed by https://github.com/shapely/shapely/pull/885
